### PR TITLE
👷 switch the pros build ci to an external repository

### PR DESCRIPTION
**Pull request by BattleCh1cken**
#### Summary

This pull request changes the pros build ci to depend on an external repository rather than being written out directly.

## Download the template for this pull request: 

> [!NOTE]  
> This is auto generated from [`Add Template to Pull Request`](https://github.com/LemLib/LemLib/actions/runs/9672028161)
- via manual download: [LemLib@0.6.0+853264.zip](https://nightly.link/LemLib/LemLib/actions/artifacts/1638404612.zip)
- via PROS Integrated Terminal: 
 ```
curl -o LemLib@0.6.0+853264.zip https://nightly.link/LemLib/LemLib/actions/artifacts/1638404612.zip;
pros c fetch LemLib@0.6.0+853264.zip;
pros c apply LemLib@0.6.0+853264;
rm LemLib@0.6.0+853264.zip;
```

### CODE REVIEW

**Comment by BattleCh1cken** [comment]
@meisZWFLZ squash and merge please

File: .github/workflows/pros-build.yml [hunk 1]
```diff
@@ -11,109 +11,16 @@ on:
...
-        run: mkdir -p template/include/lemlib/
-        
-      - name: Copy LICENSE and README.md into template
-        run: cp {LICENSE,README.md} template/include/lemlib/  
-        
-      - name: Append GitHub link to README.md
-        run: echo "\n## [Github link](${{github.server_url}}/${{github.repository}})" >> template/include/lemlib/README.md
-
-      - name: Fix relative Github links in README.md
-        run: perl -i -pe 's@(?<=[^/])(docs/assets/.*?)(?=[")])@${{github.server_url}}/${{github.repository}}/blob/master/$1?raw=true@g' template/include/lemlib/README.md
```
Comment (ABUCKY0) [inline .github/workflows/pros-build.yml:102]: We probably need to add this step back as a  step. 

```yml
# Checkout Step Before
- name: Update README.md
- run: |
  echo "\n## [Github link](${{github.server_url}}/${{github.repository}})" >> README.md
  perl -i -pe 's@(?<=[^/])(docs/assets/.*?)(?=[")])@${{github.server_url}}/${{github.repository}}/blob/master/$1?raw=true@g' README.md
# Run LemLib/pros-build@v2.0.1 after

**Review by ABUCKY0 — CHANGES_REQUESTED** [review]

File: .github/workflows/pros-build.yml (hunk 1 above)
Comment (ABUCKY0) [reply .github/workflows/pros-build.yml:102]: The only part that needs to be added back is the part that edits the readme, the action can take care of copying it

File: .github/workflows/pros-build.yml [hunk 2]
```diff
@@ -11,109 +11,16 @@ on:
...
-        with:
-          args: unzip "${{steps.template-name.outputs.name}}.zip" -d template
-        
-      - name: Create VERSION file
-        run: echo $POSTFIX > template/include/lemlib/VERSION
-        env:
-          POSTFIX: ${{ steps.template-postfix.outputs.postfix }}
+      - name: Run LemLib/pros-build
+        id: test
+        uses: LemLib/pros-build@v2.0.1
```
Comment (ABUCKY0) [inline .github/workflows/pros-build.yml:25]: This also needs 

```yml
with:
  copy_readme_and_license_to_include: true
  lib_folder_name: lemlib
 ```
 
 If you want to, you could leave this as false and do the copying in a seperate step, or you can leave it up to the action.

**Review by ABUCKY0 — CHANGES_REQUESTED** [review]

**Comment by ABUCKY0** [comment]
Oh it's gonna fail because there isn't a lemlib folder

**Comment by BattleCh1cken** [comment]
My vote would be to put this on pause until there is more work done on LemLib 0.6, and then once there's something in the repo to actually test, then we can merge this.

**Comment by ABUCKY0** [comment]
We could close this and move the changes to the v0.6-chassis-refactor branch since that's what's being worked on anyways

**Comment by BattleCh1cken** [comment]
I did what you suggested, however the workflow is still failing due to the `include` directory not being present at all in the built template, even though it exists in code and is mentioned by the `Makefile`.

I have no idea what to do to solve this, help would be appreciated.

**Comment by ABUCKY0** [comment]
It’s not because of the include folder, it’s because include/***lemlib*** doesn’t exist. You might need to push another commit to make it rerun

**Comment by BattleCh1cken** [comment]
`include/lemlib` exists, but `template/include` and `template/include/lemlib` both don't get created by `pros make template`.

I've tested this locally as well, and this seems to be the case.

**Comment by ABUCKY0** [comment]
That's not gonna do anything. I think its an issue with the Makefile

Edit: I missed the first commit, but when I tried that before it adds the makefile to the template base folder, which overwrites the user's makefile

**Comment by liam-teale** [comment]
@BattleCh1cken this still being worked on?

**Comment by BattleCh1cken** [comment]
This should be ready to merge now I think. The issue I was having before was resolved.

**Comment by liam-teale** [comment]
LGTM

**Pull request merged**